src/components/Profile/SettingsContainer/index.tsx

```diff
@@ -13,14 +13,14 @@ import SettingsBox from '../SettingsBox';
 const SettingsContainer = () => {
   const navigation = useNavigation();
   const { setAppState } = useContext(AppContext);
-  const { userState, setUserState } = useContext(UserContext);
+  const { userState, setUserState, currentHome } = useContext(UserContext);
 
   // workaround to parse data that may be undefined - to change?
   const homes = userState.homes.items;
   const homesString = JSON.stringify(homes);
   const homesJSON = JSON.parse(homesString);
-  const { home } = homesJSON[0];
-  const curHomeName = homesJSON[0].home.addressLine1.split(' ')[1];
+  const { home } = homesJSON[currentHome];
+  const curHomeName = homesJSON[currentHome].home.addressLine1.split(' ')[1];
```

Comment:
Do we still need the workaround here? It just stringifies `homes` and then parses it back into an object. I'm not sure where the potential undefined might be, maybe when it tries to destructure `const { home } = homesJSON[0]`. Would it still work if we just do `const { home } = userState.homes.items[currentHome]`?